internal/controller/bucket_controller.go

```diff
@@ -150,30 +150,37 @@ func (r BucketReconciler) reconcileDelete(ctx context.Context, objectStorageServ
 	logger.Info("Checking if bucket exists")
 	exists, err := objectStorageService.ExistsBucket(ctx, bucket)
 	if err == nil && exists {
-		logger.Info("Bucket exists, deleting")
-		err = objectStorageService.DeleteBucket(ctx, bucket)
-		if err != nil {
-			logger.Error(err, "Bucket could not be deleted")
-			return errors.WithStack(err)
-		}
-	}
-
-	logger.Info("Bucket deleted")
-
-	if bucket.Spec.AccessRole != nil && bucket.Spec.AccessRole.RoleName != "" {
-		logger.Info("Deleting bucket access role")
-		err = accessRoleService.DeleteRole(ctx, bucket)
-		if err != nil {
-			return errors.WithStack(err)
+		switch bucket.Spec.ReclaimPolicy {
+		case v1alpha1.ReclaimPolicyDelete:
+			logger.Info("Reclaim policy is set to delete, deleting bucket")
+
+			logger.Info("Bucket exists, deleting")
+			err = objectStorageService.DeleteBucket(ctx, bucket)
+			if err != nil {
+				logger.Error(err, "Bucket could not be deleted")
+				return errors.WithStack(err)
+			}
+			logger.Info("Bucket deleted")
+
+			if bucket.Spec.AccessRole != nil && bucket.Spec.AccessRole.RoleName != "" {
+				logger.Info("Deleting bucket access role")
+				err = accessRoleService.DeleteRole(ctx, bucket)
+				if err != nil {
+					return errors.WithStack(err)
+				}
+				logger.Info("Bucket access role deleted")
+			}
+		case v1alpha1.ReclaimPolicyRetain:
+			logger.Info("Reclaim policy is set to retain, not deleting bucket")
+		default:
+			logger.Info("Reclaim policy is the default one (retain), not deleting bucket")
```

Comment:
Opposite remark from @QuentinBisson here https://github.com/giantswarm/object-storage-operator/pull/180#discussion_r1673696026

We need to choose...

Both is working.
If we keep it, we don't have to recreate it if needed but we can have leftover.
If we remove it, we will have to create it again if we need it but no leftover.

Reply:
I don't follow the case where `ReclaimPolicy=Retain` and we would delete BucketCR

My thinking was:
- `ReclaimPolicy=Retain`: keep BucketCR and secret finalizer set to prevent their deletion
- `ReclaimPolicy=Delete`: remove finalizer from BucketCR and secret to have them deleted

Reply:
My comment was not the opposite but a question :)

Reply:
I thought:
ReclaimPolicy=Retain: remove finalizer from BucketCR and secret finalizer set to prevent their deletion + leave the bucket in the cloud
ReclaimPolicy=Delete: remove finalizer from BucketCR and secret to have them deleted


What is the intent between those?

Reply:
The idea behind `ReclaimPolicy=Retain` is to do nothing in case of bucket CR accidentally deleted.

Reply:
Okay then we cannot remove the finalizer